modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1525,7 +1534,89 @@ private Predicate _fillAccountEntriesPredicate(
 		Column<?, Long> column = (Column<?, Long>)table.getColumn(
 			objectField.getDBColumnName());
 
-		return column.in(ArrayUtil.toLongArray(accountEntryIds));
+		JoinStep joinStep = DSLQueryFactoryUtil.selectDistinct(
+			AccountEntryTable.INSTANCE.accountEntryId
+		).from(
+			AccountEntryTable.INSTANCE
+		);
+
+		if (_roleLocalService.hasUserRole(
+				userId, companyId, RoleConstants.ADMINISTRATOR, true)) {
+
+			return column.in(
+				joinStep.where(
+					AccountEntryTable.INSTANCE.companyId.eq(
+						companyId
+					).and(
+						AccountEntryTable.INSTANCE.status.eq(
+							WorkflowConstants.STATUS_APPROVED)
+					)));
+		}
+
+		Table<OrganizationTable> tempOrganizationTable =
+			DSLQueryFactoryUtil.select(
+				OrganizationTable.INSTANCE.companyId,
+				OrganizationTable.INSTANCE.treePath
+			).from(
+				OrganizationTable.INSTANCE
+			).innerJoinON(
+				Users_OrgsTable.INSTANCE,
+				Users_OrgsTable.INSTANCE.organizationId.eq(
+					OrganizationTable.INSTANCE.organizationId)
+			).where(
+				Users_OrgsTable.INSTANCE.userId.eq(userId)
+			).as(
+				"tempOrganizationTable", OrganizationTable.INSTANCE
+			);
+
+		return column.in(
+			joinStep.innerJoinON(
+				AccountEntryOrganizationRelTable.INSTANCE,
+				AccountEntryOrganizationRelTable.INSTANCE.accountEntryId.eq(
+					AccountEntryTable.INSTANCE.accountEntryId)
+			).where(
+				AccountEntryOrganizationRelTable.INSTANCE.organizationId.in(
```

Comment:
Is there a possibility to not have any organizationId? This would cause a SQL syntax error for this query **SELECT * FROM table WHERE field IN ()**
I'm saying that because I saw this logic [here](https://github.com/liferay/liferay-portal/blob/b69bd6a04fa27629d64448389d92bf7743e4b008/modules/apps/account/account-service/src/main/java/com/liferay/account/service/impl/AccountEntryLocalServiceImpl.java#L924).

Reply:
I was also worried about this but I tested it without any organization and it worked

Reply:
@guilhermedcamacho If instead you put a `Select` inside that returns 0 lines, it will work

```
SELECT * FROM AccountEntryOrganizationRel where organizationId in (SELECT Organization_.organizationId FROM Organization_)
```

This is why this code works

Reply:
Got it! thanks